src/main/java/net/datafaker/service/FakeValuesService.java

```diff
@@ -55,18 +57,33 @@ public FakeValuesService(Locale locale, RandomService randomService) {
         locale = normalizeLocale(locale);
 
         localesChain = localeChain(locale);
-        final List<FakeValuesInterface> all = new ArrayList(localesChain.size());
-
         for (final Locale l : localesChain) {
-            boolean isEnglish = l.equals(Locale.ENGLISH);
-            if (isEnglish) {
-                all.add(FakeValuesGrouping.getEnglishFakeValueGrouping());
+            if (l.equals(Locale.ENGLISH)) {
+                fakeValuesInterfaceMap.putIfAbsent(l, FakeValuesGrouping.getEnglishFakeValueGrouping());
             } else {
-                all.add(new FakeValues(l));
+                fakeValuesInterfaceMap.putIfAbsent(l, new FakeValues(l));
             }
         }
+    }
 
-        this.fakeValuesList = Collections.unmodifiableList(all);
+    public void addPath(Locale locale, String key, Path path) {
```

Comment:
Just curious, do you think it's possible 2 clients could possibly use the same key name here by accident? Do you think this would have a potential problem?

Reply:
The  same key is acceptable. Currently a key here is a key in `YAML` format under `faker`. 
Under the hood there will be used an internal datastructures `FakeValues` or `FakeValuesGrouping`. 
For the same key but different locale or path the internal datastructure will be different => not a problem.
In case all 3 arguments are same - it will be added only once.

P.S. probably it could be done without  `FakeValues` or `FakeValuesGrouping`, but this is kind of legacy and requires to change more code

Reply:
Thanks for clarifying

Reply:
Finally I removed key from here (it will be detected automatically while file parsing)